### PR TITLE
trainee attendance feature FN

### PR DESCRIPTION
**Pull request by bryan936M**
# PR Description

Please include a brief description of the pull request you want to raise.
- This PR is for merging this new attendance feature to the main branch.

# Description of tasks that were expected to be completed

List all the tasks and sub-tasks that were assigned to you or that you assigned yourself. Make sure you check evey completed task.
- Integration with backend API
- Re-design frontend UI
# How has this been tested?

Please describe the tests that you ran to verify your functionalities. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration.

# Number of Commits
1
The number of commits should not exceed **2 commits**. In case they are more than that, please take your time and squash them.

# Screenshots (If appropriate)
Screenshot 2023-11-17 110804

# Please check this Checklist before you submit your PR:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My code generate no warnings
- [x] My test coverage meet the set test coverage threshold
- [x] There are no vulnerabilities
- [x] There are no conflicts with the base branch

### CODE REVIEW

**Review by Dyves1 — APPROVED** [review]

**Review by irakozejules — APPROVED** [review]

**Review by Gabriel-mdv — APPROVED** [review]
great work!

**Review by ceelogre — COMMENTED** [review]
There is no attendance table.
image

File: src/components/TraineeAttendance.tsx [hunk 1]
```diff
@@ -1,15 +1,82 @@
...
+      days: day,
+      value: 'N/A',
+    }));
+
+    return {
+      weekNumber: week.weekNumber,
+      traineeAttendance: [...week.traineeAttendance, ...missingDayData],
+    };
+  });
+  console.log('optimizedWeekData:', optimizedWeekData);
```
Comment (ceelogre) [inline src/components/TraineeAttendance.tsx]: No log messages.

File: src/components/tests/TraineeAttendance.test.tsx [hunk 2]
```diff
@@ -1,46 +1,107 @@
...
+      </MockedProvider>,
+    );
+
+    await waitFor(() => screen.getByText('Monday'));
+
+    // Your logic for navigating to the next and previous pages
+    userEvent.click(screen.getByTestId('next'));
+
+    // Your assertions for the updated page content
+    expect(screen.getByText('Tuesday')).toBeInTheDocument();
```
Comment (ceelogre) [inline src/components/tests/TraineeAttendance.test.tsx:71]: Good assertion :)

File: src/components/TraineeAttendance.tsx [hunk 3]
```diff
@@ -1,15 +1,82 @@
...
 import Pagination from '../components/Pagination';
 import AttendanceData from '../dummyData/attendance.json';
 import useDocumentTitle from '../hook/useDocumentTitle';
-import Button from './Buttons';
+import { GET_WEEKLY_ATTENDANCE } from '../Mutations/Attendance';
 
-const TraineeAttendance = () => {
+interface TraineeAttendanceData {
+  getTraineeAttendanceByID: {
+    weekNumber: string;
```
Comment (ceelogre) [inline src/components/TraineeAttendance.tsx:12]: Wouldn't this be of type number logically?

Comment (bryan936M) [reply src/components/TraineeAttendance.tsx:12]: yeah, and I also found it weird. I also looked into it & found that whoever creates attendance, used a string to save week number.

**Review by ceelogre — APPROVED** [review]

**Pull request merged**